### ISSUE TITLE
Navbar Menu buttons should be  `<button>`s  (or at least have `role="button"`)

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2151]
Navbar buttons are announced by screen readers simply as 'clickable' but they should be announced as buttons. Ideally we should use the `<button>` element, but using `role="button"` would suffice.

They should also have `aria-label` set so the name of the button is announced.
